src/ClassRouteAttributes.php

```diff
@@ -79,11 +81,16 @@ public function groups(): array
                     'as' => $attributeClass->as,
                 ];
             }
-        } else {
-            $groups[] = [
-                'domain' => $this->domainFromConfig() ?? $this->domain(),
-                'prefix' => $this->prefix(),
-            ];
+        } elseif ($withDefault) {
+            $parentGroups = optional($this->parent())->groups(false);
```

Comment:
This part is a bit unreadable. Could you refactor this to new functions with a sensible name?

Reply:
@freekmurze good idea, I moved it to a new function, tell me what you think of it...